doc/tutorials/launch_uos.sh

```diff
@@ -0,0 +1,452 @@
+    fi
+else
+    boot_ipu_option="$boot_ipu_option"" -s 21,virtio-ipu "
+fi
+
+boot_cse_option=""
+if [ $cse_passthrough == 1 ]; then
+    echo "8086 5a9a" > /sys/bus/pci/drivers/pci-stub/new_id
+    echo "0000:00:0f.0" > /sys/bus/pci/devices/0000:00:0f.0/driver/unbind
+    echo "0000:00:0f.0" > /sys/bus/pci/drivers/pci-stub/bind
+    boot_cse_option="$boot_cse_option"" -s 15,passthru,0/0f/0 "
+else
+    boot_cse_option="$boot_cse_option"" -s 15,virtio-heci,0/0f/0 "
+fi
+
+#for memsize setting, total 8GB(>7.5GB) uos->6GB, 4GB(>3.5GB) uos->2GB
+memsize=`cat /proc/meminfo|head -n 1|awk '{print $2}'`
+if [ $memsize -gt 7500000 ];then
+    mem_size=6G
+elif [ $memsize -gt 3500000 ];then
+    mem_size=2G
+else
+    mem_size=512M
+fi
+
+if [ "$setup_mem" != "" ];then
+    mem_size=$setup_mem
+fi
+
+kernel_cmdline_generic="maxcpus=$2 nohpet tsc=reliable intel_iommu=off \
+   androidboot.serialno=$ser \
+   i915.enable_rc6=1 i915.enable_fbc=1 i915.enable_guc_loading=0 i915.avail_planes_per_pipe=$4 \
+   i915.enable_hangcheck=0 use_nuclear_flip=1 i915.enable_guc_submission=0 i915.enable_guc=0"
+
+boot_dev_flag=",b"
+if [ $7 == 1 ];then
+  boot_image_option="--vsbl /usr/share/acrn/bios/VSBL_debug.bin"
+else
+  boot_image_option="--vsbl /usr/share/acrn/bios/VSBL.bin"
+fi
+kernel_cmdline="$kernel_cmdline_generic"
+
+: '
+select right virtual slots for acrn_dm:
+1. some passthru device need virtual slot same as physical, like audio 0:e.0 at
+virtual #14 slot, so "-s 14,passthru,0/e/0"
+2. acrn_dm share vioapic irq between some virtual slots: like 6&14, 7&15. Need
+guarantee no virt irq sharing for each passthru device.
+FIXME: picking a virtual slot (#24 now) which is level-triggered to make sure
+audio codec passthrough working
+3. the bootable device slot is configured in compile stating in Android Guest
+image, it should be kept using 3 as fixed value for Android Guest on Gordon_peak
+ACRN project
+'
+
+#interrupt storm monitor for pass-through devices, params order: 
+#threshold/s,probe-period(s),intr-inject-delay-time(ms),delay-duration(ms)
+intr_storm_monitor="--intr_monitor 10000,10,1,100"
+
+acrn-dm --help 2>&1 | grep 'GVT args'
+if [ $? == 0 ];then
+  GVT_args=$3
+  boot_GVT_option=" -s 2,pci-gvt -G "
+else
+  boot_GVT_option=''
+  GVT_args=''
+fi
+
+ acrn-dm -A -m $mem_size -c $2$boot_GVT_option"$GVT_args" -s 0:0,hostbridge -s 1:0,lpc -l com1,stdio $npk_virt\
+   -s 9,virtio-net,$tap \
+   -s 3,virtio-blk$boot_dev_flag,/data/$5/$5.img \
+   -s 7,xhci,1-1:1-2:1-3:2-1:2-2:2-3:cap=apl \
+   -s 8,passthru,0/15/1 \
+   -s 13,virtio-rpmb \
+   -s 10,virtio-hyper_dmabuf \
+   -s 11,wdt-i6300esb \
+   $boot_audio_option \
+   $boot_cse_option \
+   $intr_storm_monitor \
+   $boot_ipu_option      \
+   --mac_seed $mac_seed \
+   -i /run/acrn/ioc_$vm_name,0x20 \
+   -l com2,/run/acrn/ioc_$vm_name \
+   $boot_image_option \
+   --enable_trusty \
+   -B "$kernel_cmdline" $vm_name
+}
+
+function launch_alios()
```

Comment:
Are you saying the open source ACRN supports AliOS now?

Reply:
The script of launching UOS for UP2 is modified from the script for MRB, and i will delete the alios part.